src/lib.rs

```diff
@@ -11,7 +11,9 @@ use std::{collections::HashMap, fmt::Display, sync::Arc};
 pub use self::{compression::Compression, row::Row};
 pub use clickhouse_derive::Row;
 
+pub mod cursor;
```

Comment:
It seems to be implementation details; let's reexport from `mod query` where this is actually used.